### PR TITLE
Add TPU performance guideline document

### PR DESCRIPTION
**Pull request by asuhan**
Section extracted from the broader documentation for cloud.

### CODE REVIEW

File: PerfGuideline.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import torch_xla_py.data_parallel as dp
+
+devices = xm.get_xla_supported_devices()
+model_parallel = dp.DataParallel(MNIST, device_ids=devices)
+                                                                                                                                                                                 
+def train_loop_fn(model, loader, device, context):
+  loss_fn = nn.NLLLoss()
+  optimizer = optim.SGD(model.parameters(), lr=lr, momentum=momentum)
+
+  for _, (data, target) in loader:
```
Comment (dlibenzi) [inline PerfGuideline.md]: Add `model.train()` above.

File: PerfGuideline.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      total_norm = (total_norm ** (1. / norm_type))
+    clip_coef = torch.tensor(max_norm, device=device) / (total_norm + 1e-6)
+    for p in parameters:
+      p.grad.data.mul_(torch.where(clip_coef < 1, clip_coef, torch.tensor(1., device=device)))
+    ```
+
+
+3.  Loops with a different number of iterations between steps are subject to similar observations as tensor shapes. PyTorch / XLA automatically handles them, but they are seen as different execution graphs and require recompilations.
+
+`print(torch_xla._XLAC._xla_metrics_report()) `can be used to print metrics at the end of each step to collect information regarding the number of compilations and operators that are part of the model but don’t have native XLA implementations.
```
Comment (jysohn23) [inline PerfGuideline.md]: nit: trailing space in `print(torch_xla._XLAC._xla_metrics_report()) `

Comment (asuhan) [reply PerfGuideline.md]: Done.

File: PerfGuideline.md (hunk 1 above)
Comment (asuhan) [reply PerfGuideline.md]: Done.

**Pull request merged**